mypy/test/teststubtest.py

```diff
@@ -1177,6 +1180,142 @@ class C: ...
+                @final
+                def eggs(self): return 42
+                @final
+                @functools.lru_cache()
+                def ham(self, obj): return obj * 2
+            """,
+            error=None,
+        )
+        # Stub methods are allowed to have @final even if the runtime doesn't...
+        yield Case(
+            stub="""
+            class E:
+                @final
+                def foo(self) -> None: ...
+                @final
+                @staticmethod
+                def bar() -> None: ...
+                @final
+                @classmethod
+                def baz(cls) -> None: ...
+                @property
+                @final
+                def eggs(self) -> int: ...
+                @final
+                def ham(self, obj: int) -> int: ...
+            """,
+            runtime="""
+            class E:
+                def foo(self): pass
+                @staticmethod
+                def bar(): pass
+                @classmethod
+                def baz(cls): pass
+                @property
+                def eggs(self): return 42
+                @functools.lru_cache()
+                def ham(self, obj): return obj * 2
+            """,
+            error=None,
+        )
+        # ...But if the runtime has @final, the stub must have it as well
+        yield Case(
+            stub="""
+            class F:
+                def foo(self) -> None: ...
+            """,
+            runtime="""
+            class F:
+                @final
+                def foo(self): pass
+            """,
+            error="F.foo",
+        )
+        yield Case(
+            stub="""
+            class G:
+                @staticmethod
+                def foo() -> None: ...
+            """,
+            runtime="""
+            class G:
+                @final
+                @staticmethod
+                def foo(): pass
+            """,
+            error="G.foo",
+        )
+        yield Case(
+            stub="""
+            class H:
+                @classmethod
+                def foo(cls) -> None: ...
+            """,
+            runtime="""
+            class H:
+                @final
+                @classmethod
+                def foo(cls): pass
+            """,
+            error="H.foo",
+        )
+        yield Case(
+            stub="""
+            class I:
+                @property
+                def foo(self) -> int: ...
+            """,
+            runtime="""
+            class I:
+                @property
+                @final
+                def foo(self): return 42
+            """,
+            error="I.foo",
+        )
+        yield Case(
+            stub="""
+            class J:
+                def foo(self, obj: int) -> int: ...
+            """,
+            runtime="""
+            class J:
+                @final
+                @functools.lru_cache()
+                def foo(self, obj): return obj * 2
```

Comment:
Let's use `...` everywhere.
```suggestion
                def foo(self, obj): ...
```

Reply:
Hmm, idk, this is the `runtime` parameter. We have several other situations in stubtest where we have trivial bodies for the `runtime` parameter. In my opinion this makes it more readable here, as otherwise it raises the question: why use `lru_cache` here?

https://github.com/python/mypy/blob/764ff6f563d0fe45c7012f6c2f1da087696e7275/mypy/test/teststubtest.py#L219-L227

https://github.com/python/mypy/blob/764ff6f563d0fe45c7012f6c2f1da087696e7275/mypy/test/teststubtest.py#L606-L634

(I don't feel too strongly about it, though -- I'll change it if you insist :)

Reply:
I generally don't have any feelings about the other peoples' code :)
So, stick to what feels best, it is totally a minor thing.